From e3adeda87d949626cb4813c30ed781800fc00a97 Mon Sep 17 00:00:00 2001
From: Dunrong Huang <riegamaths@gmail.com>
Date: Wed, 23 Dec 2015 22:05:49 +0800
Subject: [PATCH 15/23] rgw: add a new error code for invalid secret key

From http://docs.ceph.com/docs/master/radosgw/adminops/ when invalid secret
key specified, rgw should return 400 with InvalidSecretKey

Signed-off-by: Dunrong Huang <riegamaths@gmail.com>
(cherry picked from commit df6d323908baaae56a0eed898ac2ec4351450803)
Signed-off-by: Dunrong Huang <riegamaths@gmail.com>

Conflicts:
	src/rgw/rgw_user.cc
---
 src/rgw/rgw_common.h      | 1 +
 src/rgw/rgw_http_errors.h | 1 +
 src/rgw/rgw_user.cc       | 2 +-
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/rgw/rgw_common.h b/src/rgw/rgw_common.h
index 00d451a..27c4469 100644
--- a/src/rgw/rgw_common.h
+++ b/src/rgw/rgw_common.h
@@ -141,6 +141,7 @@ using ceph::crypto::MD5;
 #define ERR_USER_EXIST           2030
 #define ERR_EMAIL_EXIST          2032
 #define ERR_KEY_EXIST            2033
+#define ERR_INVALID_SECRET_KEY   2034
 #define ERR_USER_SUSPENDED       2100
 #define ERR_INTERNAL_ERROR       2200
 
diff --git a/src/rgw/rgw_http_errors.h b/src/rgw/rgw_http_errors.h
index 077eadd..8dd9c53 100644
--- a/src/rgw/rgw_http_errors.h
+++ b/src/rgw/rgw_http_errors.h
@@ -50,6 +50,7 @@ const static struct rgw_http_errors RGW_HTTP_ERRORS[] = {
     { ERR_USER_EXIST, 409, "UserAlreadyExists" },
     { ERR_EMAIL_EXIST, 409, "EmailExists" },
     { ERR_KEY_EXIST, 409, "KeyExists"},
+    { ERR_INVALID_SECRET_KEY, 400, "InvalidSecretKey"},
     { ENOTEMPTY, 409, "BucketNotEmpty" },
     { ERR_PRECONDITION_FAILED, 412, "PreconditionFailed" },
     { ERANGE, 416, "InvalidRange" },
diff --git a/src/rgw/rgw_user.cc b/src/rgw/rgw_user.cc
index eec1b6d..01596e4 100644
--- a/src/rgw/rgw_user.cc
+++ b/src/rgw/rgw_user.cc
@@ -940,7 +940,7 @@ int RGWAccessKeyPool::modify_key(RGWUserAdminOpState& op_state, std::string *err
 
   if (key.empty()) {
       set_err_msg(err_msg, "empty secret key");
-      return  -EINVAL;
+      return -ERR_INVALID_SECRET_KEY;
   }
 
   // update the access key with the new secret key
-- 
2.8.0.rc3

